src/security/Handshake.cc

```diff
@@ -504,6 +506,27 @@ Security::HandshakeParser::parseSniExtension(const SBuf &extensionData) const
     return SBuf(); // SNI extension lacks host_name
 }
 
+// RFC 8446 Section 4.2.1: SupportedVersions extension
+void
+Security::HandshakeParser::parseSupportedVersionsExtension(const SBuf &extensionData) const
+{
+    if (handshakeType == hskClientHello) {
+        Parser::BinaryTokenizer tkList(extensionData);
+        Parser::BinaryTokenizer tkVersions(tkList.pstring8("SupportedVersionsList"));
+        while (!tkVersions.atEnd()) {
+            Parser::BinaryTokenizerContext version(tkVersions, "SupportedVersion");
+            const uint16_t aVersion = tkVersions.uint16(".version");
+            if (aVersion == 0x0304)
```

Comment:
Hmm. This restriction included in "legacy_version" description of RFC  however in  "4.2.1.  Supported Versions" sections the RFC does not says anything about, but refers to a similar MUST of ServerHello:
 >  It MUST set the ServerHello.legacy_version field to  0x0303 (TLS 1.2)

I add this check too.
Also I add references to RFC and re-arranged the code a little, I hope it is better now.

Please check commit 98e8600f54

Reply:
It looks like we are using `tlsSupportedVersion` for something like "the highest supported version" but are still limiting its values to TLS v1.3. Why not TLS v1.4, etc.? Why do we need to pay so much attention to TLS v1.3 point?

Reply:
> It looks like we are using tlsSupportedVersion for something like "the highest supported version"

The tlsSupportedVersion is the highest supported version for client hello but the selected version for server hello. 
This is untill TLSv1.2. W ith TLSv1.3 required the supported_versions extension to make it clear.

> but are still limiting its values to TLS v1.3. Why not TLS v1.4, etc

The TLSv1.3 looks so strange  with al of these checks. Who knows how the TLSv1.4 will look like...

Reply:
Your answer provides useful information but does not seem to match my question. I will try to rephrase: Assuming TLS versions after v1.3 do not change how the supported version is negotiated, is there any reason to keep all these new checks in the "exactly equals to v1.3" form rather than using a more general "equals or exceeds v1.3"?

Reply:
I fixed this one.
Please check my commit c296230f42